test: integration tests for `bigquery` and some fixes

MattiasMTS opened the pull request:
# What has changed:
should be rebased after #175.

- added integration tests to the `bigquery` adapter 
- fix bigquery "first row" by refactoring the query logic
- feature -> allow setting any `QueryConfig` by using `reflect` (could be over-engineered and I am happy to discuss this) 😄 -> this is also a breaking change because of `disable-cache` -> `disable-query-cache` because it is called `DisableQueryCache` as an attribute.
- refactor `result.go` a bit based on LSP complaints.
- make sure we exit early if the utils from `testhelpers` fail

MattiasMTS commented:
@dabbertorres , I know you started this adapter - I would very much appreciate any review / second-opinions if you have time on the `reflect` usage etc. Thanks! ✨

dabbertorres commented on dbee/adapters/bigquery.go:
```diff
@@ -56,40 +63,35 @@ func (bq *BigQuery) Connect(rawURL string) (core.Driver, error) {
+			option.WithEndpoint(endpoint),
+			option.WithGRPCDialOption(grpc.WithTransportCredentials(insecure.NewCredentials())),
+			option.WithoutAuthentication(),
+			internaloption.SkipDialSettingsValidation(),
```
It'd be nice if using an "internal" package wasn't necessary here, but at least it's "only" for testing.

MattiasMTS replied:
Agreed. This is the recommended setup from testcontainer, so I will stick with that for now.

dabbertorres reviewed (commented):
The BigQuery code looks good to me overall - no major issues.

Other than the edge case I mentioned in `toKebabCase()`, my comments are primarily suggestions for tweaks, documentation, and simplification.

I can't speak to the changes to the non-BigQuery code, as it's changed quite a bit since I last contributed, and I haven't kept up-to-date on development (although I still use this plugin daily).

MattiasMTS commented on dbee/adapters/bigquery_driver.go:
```diff
@@ -227,3 +195,72 @@ func callIfSet[T any](name string, params url.Values, parse func(string) (T, err
+		fieldValue := v.Field(i)
+
+		paramName := toKebabCase(field.Name)
+		if val := params.Get(paramName); val != "" {
```
nice catch. This was one of the things making me sceptical if this "over-engineering" is just plain stupid. How likely is it that `QueryConfig` will have new attributes that need to be updated dynamically? Wdyt.

dabbertorres replied:
Probably fairly unlikely. I don't think it's changed much over time (but that can be checked).

Plus, getting to use any new fields would require updating the package dependency, so we wouldn't be able to _truly_ dynamically handle any new fields/attributes without updating _something_ anyways.

Modifying `toKebabCase()` to handle chains of uppercase letters wouldn't be too difficult though. Off the top of my head, either:
1. Use a boolean to track the case of the previous letter.
2. Check the case of the previous letter on demand.

And if the previous letter was uppercase, don't insert a `-`. Basically just another `&& condition` to your existing `if`.

Adding in a tabled unit test for it definitely can't hurt.

MattiasMTS commented:
thanks for the comments and feedback @dabbertorres ! I have addressed most of them now and also cleaned up a bit of code. Feel free to take a final look if you want, otherwise I plan to merge this as it is.
